org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/IClassContent.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.core.compiler;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+
+/**
+ * Encapsulates the produced class of the compile operation
+ * @since 3.33
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+public interface IClassContent {
+
+	/**
+	 * @return the class bytes of this class content or <code>null</code> if no content was written
+	 * @throws CoreException if gathering the bytes failed
+	 */
+	byte[] getBytes() throws CoreException;
+
```

Comment:
To me, readBytes() sounds better if reading from a stream is involved here. It makes it more apparent of what happens.

Reply:
The content is not necessarily read from a stream, it might be cached or gathers from somewhere else (see setBytes), so probably I should adjust the javadoc to not contain "written" but "is not available"